src/schema/posts.ts

```diff
@@ -2768,29 +2768,81 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       _,
       { postId }: { postId: string },
       ctx: AuthContext,
-    ): Promise<GQLEmptyResponse> => {
+    ): Promise<TransactionCreated> => {
+      const { userId } = ctx;
       const post = await validatePostBoostPermissions(ctx, postId);
       const campaignId = post?.flags?.campaignId;
 
       if (!campaignId) {
         throw new ValidationError('Post is not currently boosted');
       }
 
-      await ctx.con.transaction(async (entityManager) => {
+      const result = await ctx.con.transaction(async (entityManager) => {
+        const { currentBudget } = await skadiApiClient.cancelPostCampaign({
+          campaignId,
+          userId: ctx.userId,
+        });
+
         await entityManager
           .getRepository(Post)
           .update(
             { id: postId },
             { flags: updateFlagsStatement<Post>({ campaignId: null }) },
           );
 
-        await skadiApiClient.cancelPostCampaign({
-          campaignId,
-          userId: ctx.userId,
-        });
+        const toRefund = parseFloat(currentBudget);
+
+        const userTransaction = await entityManager
+          .getRepository(UserTransaction)
+          .save(
+            entityManager.getRepository(UserTransaction).create({
+              id: randomUUID(),
+              processor: UserTransactionProcessor.Njord,
+              receiverId: userId,
+              status: UserTransactionStatus.Success,
+              productId: null,
+              senderId: systemUser.id,
+              value: usdToCores(toRefund),
+              valueIncFees: 0,
+              fee: 0,
+              flags: { note: 'Post boost canceled' },
+              referenceId: campaignId,
+              referenceType: UserTransactionType.PostBoost,
+            }),
+          );
+
+        try {
+          const transfer = await transferCores({
+            ctx: { userId },
+            transaction: userTransaction,
+            entityManager,
+          });
+
+          return {
+            transfer,
+            transaction: {
+              referenceId: campaignId,
+              transactionId: userTransaction.id,
+              balance: {
+                amount: parseBigInt(transfer.receiverBalance?.newBalance),
+              },
+            },
+          };
+        } catch (error) {
+          if (error instanceof TransferError) {
+            await throwUserTransactionError({
```

Comment:
Just to check this will also rollback `campaignId` on Post if transaction fails, you ok with that?

Reply:
then you need to handle that manually in catch block

Reply:
I mean, we should not rollback the campaign id, and keep it as it is. So that when the user tries to send another request, the result would be as intended. Where the campaign is already not boosted, which is true.

Reply:
ok 👍